compiler-plugin/src/main/scala/com/trueaccord/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -1074,6 +1074,7 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
       s"""object $className extends $companionType {
          |  implicit def messageCompanion: $companionType = this""")
       .indent
+      .add(s"type ValueType = $className")
```

Comment:
Why not just type `type ValueType = A` in the  `GeneratedMessageCompanion` trait analogously to what's done for enums?